coil-base/src/main/java/coil/request/Parameters.kt

```diff
@@ -62,14 +85,14 @@ class Parameters private constructor(
 
     class Builder {
 
-        private val map: SortedMap<String, Entry>
+        private val map: MutableMap<String, Entry>
```

Comment:
Why was this sorted before?

Reply:
@Jawnnypoo The iteration order had to be consistent irrespective of insertion order so the cache key would match up. If we didn't sort by keys, the parameters would produce two difference cache keys. E.g.

```kotlin
val parameters = Parameters.Builder()
    .set("key1", "first_value")
    .set("key2", "second_value")
    .build()
// would produce the string cache key "key1=first_value#key2=second_value"

val parameters = Parameters.Builder()
    .set("key2", "second_value")
    .set("key1", "first_value")
    .build()
// would produce the string cache key "key2=second_value#key1=first_value"
```

One of the gotchas of using a string cache key. Now order doesn't matter since `Parameters` are compared using `Map.equals`!

Reply:
Ahhhh gotcha. Thanks for clarifying!